libkbfs/random_backoff.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2016 Keybase Inc. All rights reserved.
+// Use of this source code is governed by a BSD
+// license that can be found in the LICENSE file.
+
+package libkbfs
+
+import (
+	"math/rand"
+	"sync"
+	"time"
+)
+
+var randomBackoffRand *rand.Rand
+
+func init() {
+	randomBackoffRand = rand.New(rand.NewSource(time.Now().Unix()))
+}
+
+type fireOnce struct {
+	ch   chan struct{}
+	once sync.Once
+}
+
+func newFireOnce() *fireOnce {
+	return &fireOnce{ch: make(chan struct{})}
+}
+
+func (o *fireOnce) fire() {
+	if o == nil {
```

Comment:
hmm.. I guess we can change `fireOnce` to be a value type, and their fields to be pointer types, and check pointers in its methods (so that we can still do stuff like `b.swap(f).fire()`)